smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServerCommandGenerator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.typescript.codegen;
+
+import static software.amazon.smithy.typescript.codegen.CodegenUtils.getBlobStreamingMembers;
+import static software.amazon.smithy.typescript.codegen.CodegenUtils.writeStreamingMemberType;
+
+import java.util.List;
+import java.util.Optional;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.OperationIndex;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+
+/**
+ * Generates a client command using plugins.
+ */
+final class ServerCommandGenerator implements Runnable {
+
+    private final TypeScriptSettings settings;
+    private final Model model;
+    private final OperationShape operation;
+    private final SymbolProvider symbolProvider;
+    private final TypeScriptWriter writer;
+    private final OperationIndex operationIndex;
+    private final Symbol inputType;
+    private final Symbol outputType;
+
+    ServerCommandGenerator(
+            TypeScriptSettings settings,
+            Model model,
+            OperationShape operation,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer
+    ) {
+        this.settings = settings;
+        this.model = model;
+        this.operation = operation;
+        this.symbolProvider = symbolProvider;
+        this.writer = writer;
+
+        Symbol operationSymbol = symbolProvider.toSymbol(operation);
+        operationIndex = OperationIndex.of(model);
+        inputType = operationSymbol.expectProperty("inputType", Symbol.class);
+        outputType = operationSymbol.expectProperty("outputType", Symbol.class);
+    }
+
+    @Override
+    public void run() {
+        addInputAndOutputTypes();
+    }
+
+    private void addInputAndOutputTypes() {
+        writeInputType(inputType.getName(), operationIndex.getInput(operation));
+        writeOutputType(outputType.getName(), operationIndex.getOutput(operation));
+        writer.write("");
+    }
+
+    // TODO: flip the these
```

Comment:
Should this get removed? Updated with additional content?  Seems incomplete.

Reply:
See #284 for the origin of this change.